replica/table.cc

```diff
@@ -644,26 +669,38 @@ table::add_sstables_and_update_cache(const std::vector<sstables::shared_sstable>
 future<>
 table::update_cache(compaction_group& cg, lw_shared_ptr<memtable> m, std::vector<sstables::shared_sstable> ssts) {
     auto permit = co_await seastar::get_units(_sstable_set_mutation_sem, 1);
-    mutation_source_opt ms_opt;
-    if (ssts.size() == 1) {
-        ms_opt = ssts.front()->as_mutation_source();
-    } else {
-        std::vector<mutation_source> sources;
-        sources.reserve(ssts.size());
-        for (auto& sst : ssts) {
-            sources.push_back(sst->as_mutation_source());
+
+    class updater : public table_sstables_adder {
+        lw_shared_ptr<memtable> _m;
```

Comment:
Can you explain why we need this tall hierarchy?

Reply:
vs. what?  open coding the `table_sstables_adder` in this local `updater`?